tests/unit/services/loading-test.ts

```diff
@@ -1,49 +1,145 @@
 import { module, test } from 'qunit';
 import { setupTest } from 'ember-qunit';
 import { timeout } from 'ember-concurrency';
+import LoadingService from 'ember-loading/services/loading';
+import { defer } from 'rsvp';
 
 module('Unit | Service | loading', function(hooks) {
   setupTest(hooks);
 
-  test('switching on/off is behaving properly', function(assert) {
-    let service = this.owner.lookup('service:loading');
+  test('can run async jobs', async function(assert) {
+    let deferred = defer();
+    let called = false;
+    let job = () => {
+      called = true;
+      return deferred.promise;
+    };
+
+    let service: LoadingService = this.owner.lookup('service:loading');
     assert.notOk(service.get('isLoading'));
-    service.start();
+    assert.notOk(service.get('showLoading'));
+
+    let promise = service.run(job);
+    assert.ok(called, 'job has been called');
     assert.ok(service.get('isLoading'));
-    service.stop();
-    assert.notOk(service.get('isLoading'));
-  });
+    assert.ok(service.get('showLoading'));
 
-  test('stays on when multiple requests are requested and switch off while all requests are done', function(assert) {
-    let service = this.owner.lookup('service:loading');
+    deferred.resolve();
+    await promise;
     assert.notOk(service.get('isLoading'));
-    service.start();
-    service.start();
-    assert.ok(service.get('isLoading'));
-    service.stop();
-    assert.ok(service.get('isLoading'));
-    service.stop();
+    assert.ok(service.get('showLoading'));
+
+    await timeout(1);
     assert.notOk(service.get('isLoading'));
+    assert.notOk(service.get('showLoading'));
```

Comment:
There are multiple assertions of the same type per test. In case one fails, you'll receive something like `expected true to be false` and will have to use the inspector to figure out the cause.

I prefer to add assertion messages like these:

```
#0 Initial: isLoading
#0 Initial: showLoading
#1 After starting the first job: isLoading
#1 After starting the first job: showLoading
#2 After starting the second job: isLoading
#2 After starting the second job: showLoading
#3 After finishing the first job: isLoading
#3 After finishing the first job: showLoading
#4 After finishing the second job: isLoading
#4 After finishing the second job: showLoading
```

Reply:
Yes, indeed. Was a bit lazy yesterday night at 1am or so :wink: